bika/lims/content/analysisrequest.py

```diff
@@ -1916,7 +1916,7 @@ def getRequestID(self):
         return self.getId()
 
     @deprecated('[1703] Use setId(new_id) instad')
-    def setRequestID(self, new_id):
+    def setRequestID(self, new_id, **kwargs):
```

Comment:
I see this method is tagged as `deprecated` . I think would be better to find out where this function is called and replace the call by the one suggested in the deprecation message `setId(new_id)`

Reply:
since `RequestID` is a field in AR Schema, it is called by default while an AR object is being created. We do not call it anywhere from code currently.

Reply:
Indeed, the AR create process passes all field values to `bika.lims.utils.analysisrequest.create_analysisrequest`: 
https://github.com/senaite/bika.lims/blob/senaite-integration/bika/lims/browser/analysisrequest/add2.py#L1748
which calls `processForm` method of the AR:
https://github.com/senaite/bika.lims/blob/senaite-integration/bika/lims/utils/analysisrequest.py#L75
As far as I know, this looks up the default `mutator` of the field and calls it with the given value. So we have to probably overwrite that default mutator in the Schema